adserver/forms.py

```diff
@@ -1150,6 +1157,190 @@ class Meta:
         }
 
 
+class BulkAdvertisementUploadCSVForm(forms.Form):
+    """
+    Used by advertisers to upload ads in bulk.
+
+    The actual saving of bulk ads is done by the BulkAdvertisementPreviewForm.
+    """
+
+    REQUIRED_FIELD_NAMES = [
+        "Name",
+        "Live",
+        "Link URL",
+        "Image URL",
+        "Headline",
+        "Content",
+        "Call to Action",
+    ]
+
+    advertisements = forms.FileField(
+        label=_("Advertisements"), help_text=_("Upload a CSV using our ad template")
+    )
+
+    def __init__(self, *args, **kwargs):
+        """Add the form helper and customize the look of the form."""
+        if "flight" in kwargs:
+            self.flight = kwargs.pop("flight")
+        else:
+            raise RuntimeError("'flight' is required for the bulk ad form")
+
+        super().__init__(*args, **kwargs)
+
+        self.fields["advertisements"].widget.attrs["accept"] = "text/csv"
+
+        self.helper = FormHelper()
+        self.helper.attrs = {
+            "id": "advertisements-bulk-upload",
+            "enctype": "multipart/form-data",
+        }
+
+        self.helper.layout = Layout(
+            Fieldset(
+                "",
+                Field("advertisements", placeholder="Upload a CSV file"),
+                css_class="my-3",
+            ),
+            Submit("submit", _("Preview ads")),
+        )
+
+    def clean_advertisements(self):
```

Comment:
It feels like this validation logic is likely also used in the ad form -- does it make sense to combine it into a util function?

Reply:
Ya, makes sense.

Reply:
I moved the image/text validation onto the AdType model.